src/nupic/regions/tm_region.py

```diff
@@ -805,7 +805,22 @@ def writeToProto(self, proto):
       self.computePredictedActiveCellIndices)
     proto.orColumnOutputs = self.orColumnOutputs
 
-    self._tfdr.write(proto.temporalMemory)
+    if self.temporalImp == "py":
+      tmProto = proto.init("backtrackingTM")
+      self._tfdr.write(tmProto)
+    elif self.temporalImp == "cpp":
+      tmProto = proto.init("backtrackingTMCpp")
+      self._tfdr.write(tmProto)
+    elif self.temporalImp == "tm_py":
+      tmProto = proto.init("temporalMemory")
+      self._tfdr.write(tmProto)
+    elif self.temporalImp == "tm_cpp":
+      tmProto = proto.init("temporalMemory")
```

Comment:
I believe the Python TemporalMemory uses the TemporalMemory.capnp file from nupic.bindings